showcase/public/ellipsizable-text.js

```diff
@@ -0,0 +1,97 @@
+const template = Object.assign(document.createElement('template'), {
+  innerHTML: `
+      <style>
+        .text {
+          white-space: nowrap;
+        }
+  
+        .text--overflown {
+          cursor: pointer;
+          text-overflow: ellipsis;
+          overflow: hidden;
+          display: block;
+        }
```

Comment:
```css
        .text {
          white-space: nowrap;
          text-overflow: ellipsis;
          overflow: hidden;
          display: block;
        }
  
        .text--overflown {
          cursor: pointer;
        }
```

Reply:
we can move some of these properties but not all, if I do that then when I get the width it won't be more than the parent's